src/safeposix/filesystem.rs

```diff
@@ -87,15 +91,36 @@ impl FilesystemMetadata {
             atime: time, ctime: time, mtime: time,
             filename_to_inode_dict: init_filename_to_inode_dict(ROOTDIRECTORYINODE)};
         retval.inodetable.insert(ROOTDIRECTORYINODE, Inode::Dir(dirinode));
+
         retval
     }
 }
 
-pub fn load_fs_special_files() {
+pub fn load_fs() {
+
+    // Create initial cage, probably will move this
     let utilcage = Cage{cageid: 0,
-                        cwd: interface::RustLock::new(interface::RustRfc::new(interface::RustPathBuf::from("/"))),
-                        parent: 0, 
-                        filedescriptortable: interface::RustLock::new(interface::RustHashMap::new())};
+        cwd: interface::RustLock::new(interface::RustRfc::new(interface::RustPathBuf::from("/"))),
+        parent: 0, 
+        filedescriptortable: interface::RustLock::new(interface::RustHashMap::new())};
+
+    let metadatafo = interface::openfile(METADATAFILENAME.to_string(), true);
+    
+    // If the metadata file exists, just close the file for later restore
+    // If it doesn't, lets create a new one and persist it.
+    match metadatafo {
+        Ok(file) => {file.close().unwrap();}
+        Err(_e) => {FilesystemMetadata::blank_fs_init(); persist_metadata(None);}
+    };
```

Comment:
I think this means metadata file object but maybe making it metadata_fileobj would make that clearer than fo.
Also, is the persist necessary right after the blank_fs_init? I thought we discussed it and said it wasn't.

Reply:
I think persist here does make sense, we basically init the metadata file and also give us something to do the "restore" from at the next step that is consistent whether or not we did a blank init.